src/m365/spo/commands/tenant/tenant-commandset-set.ts

```diff
@@ -0,0 +1,242 @@
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const appCatalogUrl = await spo.getTenantAppCatalogUrl(logger, this.debug);
+      if (!appCatalogUrl) {
+        throw 'No app catalog URL found';
+      }
+
+      await this.updateTenantWideExtension(appCatalogUrl, args.options, logger);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getListItemById(logger: Logger, weburl: string, listServerRelativeUrl: string, id: string): Promise<any> {
+    if (this.verbose) {
+      logger.logToStderr(`Getting the list item by id ${id}`);
+    }
+    const reqOptions: CliRequestOptions = {
+      url: `${weburl}/_api/web/GetList('${formatting.encodeQueryParameter(listServerRelativeUrl)}')/Items(${id})`,
+      headers: {
+        'accept': 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    const listItemInstance: any = await request.get<any>(reqOptions);
+
+    return listItemInstance;
+  }
+
+  private async updateTenantWideExtension(appCatalogUrl: string, options: Options, logger: Logger): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr('Updating tenant wide extension to the TenantWideExtensions list');
+    }
+
+    const listServerRelativeUrl: string = urlUtil.getServerRelativePath(appCatalogUrl, '/lists/TenantWideExtensions');
+    const listItem = await this.getListItemById(logger, appCatalogUrl, listServerRelativeUrl, options.id);
+
+    if (listItem.TenantWideExtensionLocation.indexOf("ClientSideExtension.ListViewCommandSet") === -1) {
+      throw 'The item is not a ListViewCommandSet';
+    }
+
+    const formValues: any = [];
+    if (options.newTitle !== undefined) {
+      formValues.push({
+        FieldName: 'Title',
+        FieldValue: options.newTitle
+      });
+    }
+
+    if (options.clientSideComponentId !== undefined) {
+      formValues.push({
+        FieldName: 'TenantWideExtensionComponentId',
+        FieldValue: options.neclientSideComponentIdwTitle
+      });
+    }
+
+    if (options.location !== undefined) {
+      formValues.push({
+        FieldName: 'TenantWideExtensionLocation',
+        FieldValue: this.getLocation(options.location)
+      });
+    }
+
+    if (options.listType !== undefined) {
+      formValues.push({
+        FieldName: 'TenantWideExtensionListTemplate',
+        FieldValue: this.getListTemplate(options.listType)
+      });
+    }
+
+    if (options.clientSideComponentProperties !== undefined) {
+      formValues.push({
+        FieldName: 'TenantWideExtensionComponentProperties',
+        FieldValue: options.clientSideComponentProperties
+      });
+    }
+
+    if (options.webTemplate !== undefined) {
+      formValues.push({
+        FieldName: 'TenantWideExtensionWebTemplate',
+        FieldValue: options.webTemplate
+      });
+    }
+
+    const data = {
```

Comment:
Let's create this object from the requestObject directly. No use to assign it to a variable here